nixos/modules/services/mail/smtp-to-telegram.nix

```diff
@@ -0,0 +1,113 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.smtp-to-telegram;
+
+in
+
+{
+  meta = {
+    maintainers = with maintainers; [ patryk27 ];
+  };
+
+  ###### interface
+
+  options = {
+    services.smtp-to-telegram = {
+      enable = mkEnableOption "SMTP server that forwards messages to Telegram";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.smtp-to-telegram;
+        defaultText = "pkgs.smtp-to-telegram";
+        description = "Which smtp-to-telegram derivation to use.";
+      };
+
+      messageTemplate = mkOption {
+        type = types.str;
+        default = "From: {from}\\nTo: {to}\\nSubject: {subject}\\n\\n{body}";
+        description = ''
+          Template used to convert e-mails into Telegram messages.
+
+          Available variables:
+          <literal>from</literal>,
+          <literal>to</literal>,
+          <literal>subject</literal>,
+          <literal>body</literal>.
+        '';
+      };
+
+      smtpListen = mkOption {
+        type = types.str;
+        default = "127.0.0.1:2525";
+        description = "Address at which the SMTP server will be listening.";
+      };
+
+      smtpPrimaryHost = mkOption {
+        type = types.str;
+        default = config.networking.hostName;
+        description = "Hostname at which the SMTP server will be listening.";
+      };
+
+      telegramApiPrefix = mkOption {
+        type = types.str;
+        default = "https://api.telegram.org/";
+        description = "Telegram API's URL; must end with a slash.";
+      };
+
+      telegramBotToken = mkOption {
```

Comment:
This will be placed in plain text in the nix store which is publicly readable. We should use a secret file instead.